pkg/kwok/controllers/node_controller.go

```diff
@@ -615,3 +632,21 @@ func (c *NodeController) funcNodeName() string {
 func (c *NodeController) funcNodePort() int {
 	return c.nodePort
 }
+
+// backoffDelay calculates the backoff delay of a Stage job based on its failed times
+func (c *NodeController) backoffDelay(failedTimes int) time.Duration {
```

Comment:
It feels like this function is moving to utils and will be shared for node, pod and stage controllers.